cli/defenseclaw/commands/cmd_doctor.py

```
@@ -310,20 +368,32 @@ def _resolve_api_key(env_name: str, dotenv_path: str) -> str:
     """Resolve an API key from env → .env file → empty."""
     val = os.environ.get(env_name, "")
     if val:
+        data_dir = os.path.dirname(os.path.abspath(dotenv_path))
+        if credential_provenance.was_injected_from_dotenv(data_dir, env_name, val):
+            if _gateway_dotenv_safety_problem(type("_DotenvScope", (), {"data_dir": data_dir})()):
```

Ruff (0.16.0)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)